packages/@aws-cdk/aws-codebuild/README.md

```diff
@@ -80,6 +80,15 @@ const project = new codebuild.Project(this, 'MyProject', {
 }
 ```
 
+You can also add the Project to the Pipeline directly:
+
+```ts
+// equivalent to the code above:
+project.addBuildToPipeline(buildStage, 'CodeBuild', {
```

Comment:
Why not just `addToPipeline`? It's a build project after all...
If this is because a build project can be added as a test action, this can be indicated in a prop

Reply:
You're correct Elad - the reason is that there will also be a test Action in the future (`PipelineTestAction`) that uses CodeBuild Projects.

My justification for separating them is that `PipelineBuildAction` and `PipelineTestAction` may actually have different construction properties (the test Action doesn't produce outputs, for example). Separation into `addBuildToPipeline` and `addTestToPipeline` gives us the flexibility to have these different props. If we have one `addToPipeline` method with a `type`: `ProjectType.BUILD/.TEST` property, we will forever be locked into them having the same properties.

Does this make sense?

Reply:
Actually, we simply can't express `addToPipeline` at all now - this method would have to return either the `PipelineBuildAction` or the `PipelineTestAction` type, and there is no way of expressing that without using generics, which JSII does not support.

Reply:
They could share a base class that has a disambiguator property. But that's not so great developer experience...